apps/govern/components/Contracts/EditVotes.tsx

```diff
@@ -0,0 +1,409 @@
+          onChange={(value) => {
+            if (value) {
+              setAllocation(value, index);
+            }
+          }}
+        />
+        <Button
+          icon={<DeleteOutlined />}
+          onClick={() => removeAllocation(index)}
+          style={{ flex: 'none' }}
+        />
+      </Flex>
+    ),
+    width: 140,
+  },
+];
+
+const ConfirmModal = ({
+  isOpen,
+  handleOk,
+  handleClose,
+  isLoading,
+  allocationsLength,
+  totalPower,
+}: {
+  isOpen: boolean;
+  handleOk: () => void;
+  handleClose: () => void;
+  isLoading: boolean;
+  allocationsLength: number;
+  totalPower: number;
+}) => {
+  return (
+    <Modal
+      title="Confirm voting weight update"
+      open={isOpen}
+      onOk={handleOk}
+      onCancel={handleClose}
+      cancelText="Cancel"
+      okText="Confirm voting weight"
+      confirmLoading={isLoading}
+    >
+      <Paragraph>
+        {`You're allocating ${parseFloat(
+          (totalPower / 100).toFixed(2),
+        )}% of your voting power to ${allocationsLength} staking contracts.`}
+      </Paragraph>
+      <Paragraph>
+        {`After you confirm, you'll enter a 10 day cooldown period. You won't be able to update your weights during that time.`}
+      </Paragraph>
+
+      {totalPower < 10_000 && (
+        <Alert
+          // TODO: add blue info alerts as in Pearl
+          className="mb-16"
+          message={`${parseFloat(
+            ((10_000 - totalPower) / 100).toFixed(2),
+          )}% of your voting power is unallocated - this will be applied to the Rollover Pool and may be used in future epochs.`}
+          showIcon
+        />
+      )}
+    </Modal>
+  );
+};
+
+export const EditVotes = ({ allocations, setAllocations, setIsUpdating }: EditVotesProps) => {
+  const dispatch = useAppDispatch();
+  const { address: account } = useAccount();
+  const { userVotes, stakingContracts } = useAppSelector((state) => state.govern);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onCancel = () => setAllocations([]);
+
+  const updateAllocation = (value: number, index: number) => {
+    setAllocations((prev) => {
+      const newAllocations = [...prev];
+      newAllocations[index] = { ...newAllocations[index], weight: value };
+      return newAllocations;
+    });
+  };
+
+  const removeAllocation = (index: number) => {
+    setAllocations((prev) => prev.filter((_, i) => i !== index));
+  };
+
+  const totalPower = allocations.reduce((sum, item) => sum + item.weight * 100, 0);
+  const isError = totalPower > 10_000;
+
+  const showModal = () => setIsModalOpen(true);
+  const closeModal = () => setIsModalOpen(false);
+
+  const updateVotingWeight = useCallback(async () => {
+    if (!account) return;
+    if (totalPower > 10_000) return;
+
+    setIsLoading(true);
+
+    // Check if any of the nominees were removed from voting
+    const removedNominees = await checkIfNomineeRemoved(allocations);
```

Comment:
might be able to batch the await calls in this function with Promise.allSettled ?

Reply:
it's enough just to get the list of removedNominees (stores in the contract's state) and check if any of the nominee, that user votes on, is in the list. It'd sone inside the `checkIfNomineeRemoved` method